api/app/references/api.py

```diff
@@ -130,12 +131,37 @@ def post(self):
         token = args['token']
         uploaded_document = args['uploaded_document']
         reference_request = reference_request_repository.get_by_token(token)
+        event = event_repository.get_event_by_response_id(reference_request.response_id)
+
         if not reference_request:
             return REFRERENCE_REQUEST_WITH_TOKEN_NOT_FOUND
 
         if reference_request.has_reference():
             return DUPLICATE_REFERENCE_SUBMISSION
-        
+
+        if not event.is_application_open:
+            return APPLICATIONS_CLOSED
+
         reference = Reference(reference_request_id=reference_request.id, uploaded_document=uploaded_document)
         reference_request_repository.add(reference)
         return {}, 201
+
+    def put(self):
+        args = self.put_req_parser.parse_args()
+        token = args['token']
+        uploaded_document = args['uploaded_document']
+        reference_request = reference_request_repository.get_by_token(token)
+        event = event_repository.get_event_by_response_id(reference_request.response_id)
+        reference = reference_request_repository.get_reference_by_reference_request_id(reference_request.id)
+
+        if not reference_request:
+            return {}, 404
```

Comment:
Perhaps in the api\app\utils\errors.py file add a new error `REFERENCE_REQUEST_NOT_FOUND = ({'message': 'No response found for the given event and user'}, 404)` and then import that and use it here

Reply:
Added new error and used it in the reference API.